torch/distributions/constraint_registry.py

```diff
@@ -153,13 +153,21 @@ def __call__(self, constraint):
 ################################################################################
 
 @biject_to.register(constraints.real)
-@biject_to.register(constraints.real_vector)
 @transform_to.register(constraints.real)
-@transform_to.register(constraints.real_vector)
 def _transform_to_real(constraint):
     return transforms.identity_transform
 
 
+@biject_to.register(constraints.independent)
+def _biject_to_independent(constraint):
+    return biject_to(constraint.base_constraint)
```

Comment:
I'm not sure if this will be enough to address #50496 since the event dim information will be lost when we call `log_abs_det_jacobian`, i.e. it seems to me that the transform returned for both the independent as well as the base constraint is the same. One way would be to post-hoc modify the output event dim of a transform to handle that use case. @feynmanliang - please correct me if I missed something.

Reply:
I just think that we no longer need `input_event_dim`, `output_event_dim` in transforms. All we need is to define a correct `domain`, `codomain`. For composed transform, we still need to handle the logic properly.

Reply:
Interesting, yes that's one of the differences w.r.t. numpyro. That's probably worth another discussion.

Reply:
@fritzo - I like @fehiepsi's suggestion of modifying the domain/codomain's event dim (and remove input/output event dims since that wasn't part of last release) to handle this, but regardless, all the proposed changes in this PR look great to me, so please feel free to defer this to later.

Reply:
done

Reply:
That sounds right. I think we can follow @feynmanliang's suggestion in a future PR and say wrap with an `IndependentTransform` or set the `Transform.event_dim`. I'll demote this PR from "Fixes" to "Addresses".